[DOCS] Reworks some parts of EMM API docs (#54872)
elastic/elasticsearch: docs/reference/ml/anomaly-detection/apis/estimate-model-memory.asciidoc

<titleabbrev>Estimate model memory</titleabbrev>
++++

> Makes an estimation on the model memory an {anomaly-job} is likely to need. It 

Reviewer: Not sure if this is clearer or not?:
```suggestion
Makes an estimation of the memory usage for an {anomaly-job} model. It 
```

> is based on analysis configuration details and cardinality estimates for the 

Reviewer: ```suggestion
is based on analysis configuration details for the job and cardinality estimates for the 
```
